include/votca/csg/triclinicbox.h

```diff
@@ -29,12 +29,16 @@ class TriclinicBox : public BoundaryCondition {
   Eigen::Vector3d BCShortestConnection(
       const Eigen::Vector3d &r_i, const Eigen::Vector3d &r_j) const override;
 
-  eBoxtype getBoxType() override { return typeTriclinic; }
+  virtual std::unique_ptr<BoundaryCondition> Clone() const override {
```

Comment:
I am not sure why we need a clone method, if you want something like that write a copyconstructor and assignment operator.

Reply:
Sounds good

Reply:
we need a move and move assignment as well then.

Reply:
Actually, I was looking over this I'm not sure copy is appropriate, the reason for returning a unique_ptr to the base class allows you to dynamically store the child classes in a vector of base class pointers. I will keep it as is though if you think there is a better name for the method I'm open for ideas?

Reply:
Why would you not have a copy constructor then?

Reply:
Because you are not explicitly copying the object, the object is always being handled by pointers. Why bother adding code if you are just manipulating the pointers. The clone method is returning a unique pointer to the base class when it is called from a child class, even though it is copying the object that is not all it is doing, using a copy constructor would be misleading in this case.

Reply:
okay I see your point.